xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm

```diff
@@ -120,20 +120,22 @@
     ##   #if($!tags.size() == 0)
     ##     No tags
     ##   #else
-    $services.localization.render('core.tags.list.label')
-    #foreach($tag in $tags)
-        #displayTag($tag)
-    #end
     ##   #end
-    #if($hasedit)
-        #if($xwiki.tag)
-          <div class="tag-tool tag-add">#if("$!{request.showTagAddForm}" == '')<a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>#else #displayAddForm()#end</div>
-        #else
-            ## TODO
+    <div class="doc-tags" id="${tagsId}">
```

Comment:
AFAIU you don't want to remove this div to avoid a CSS issue, shown in https://forum.xwiki.org/t/new-solution-for-xwiki-17588-issue/8526/4 I'm ok with that, but not sure why you don't use the a hidden class on it instead of displaying the label in the div only if there's some tags. IMO that's better to rely on a hidden div containing the right label: the semantic is preserved and it's easier to understand the HTML. Note that we have a dedicated "hidden" CSS class.